PrestaShop/PrestaShop
# Migrate merchandise return view - base

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR migrates "merchandise return" view/edit BO page to symfony. This PR migrates only view of merchandise return **without product list or label printing** as planned in roadmap on https://github.com/PrestaShop/PrestaShop/issues/24421.
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | One item of https://github.com/PrestaShop/PrestaShop/issues/24421
| How to test?  | See below

## How to test

1. Enable merchandise returns option
2. Create some merchandise returns.
3. Go to `admin-dev/index.php/sell/customer-service/merchandise-return/` which is the list. Click on any of existing returns to open the page.
4. In this page, you should be able to change the 'order return state' and save it successfully. The design is not very good because the page is not complete.

⚠️ **PDF printing and product list are not included in this PR.** They will be implemented later. Only changing the state is implemented here. The design is not very good because the page is not complete. It will look so much better when next PRs are submitted and merged.

## About class and namespace naming

All new classes and namespaces in this PR are `OrderReturn` instead of `MerchandiseReturn` because that's the direction we decided to go with.

Everything codewise in legacy was called OrderReturn as well so we want to maintain codebase consistency. So OrderReturn in code = MerchandiseReturn in interface.

There will be a separate PR to change all existing code(mostly related to Merchandise Return grid) to reflect that (see roadmap https://github.com/PrestaShop/PrestaShop/issues/24421).



---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/19420)
